src/rashdf/geom.py

```diff
@@ -43,13 +48,97 @@ def mesh_cell_polygons(self) -> GeoDataFrame:
         GeoDataFrame
             A GeoDataFrame containing the 2D flow mesh cell polygons.
         """
-        raise NotImplementedError
+        # recursive coord cleaning
+        def clean_coords(end_coord: list = None) -> None:
+            if not end_coord:
+                target = raw_coords.pop(0)
+            else:
+                i = -1
+                for face in raw_coords:
+                    i+=1
+                    if end_coord == face[0]:
+                        target = raw_coords.pop(i)
+                    elif end_coord == face[-1]:
+                        target = raw_coords.pop(i)
+                        target=list(reversed(target))
+            [cleaned_coords.append(coord) for coord in target]
+            if raw_coords: clean_coords(target[-1])
+        def polygonize(face_coord_list) -> Polygon:
+            global cleaned_coords; global raw_coords
+            cleaned_coords=list(); raw_coords=face_coord_list
+            clean_coords()
+            return Polygon(cleaned_coords)
+
+        # assemble face dict
+        face_df = self.mesh_cell_faces()
+        face_df["face_code"] = face_df["face_id"].astype(str)+face_df["mesh_name"]
+        face_df["coords"] = face_df["geometry"].apply(lambda face: face.coords)
+        face_dict = face_df[["face_code", "coords"]].set_index("face_code", inplace=False).to_dict()["coords"]
+
+        # assemble cell poly df
+        cell_dict = {"mesh_name":[], "cell_id":[], "face_id_list":[]}
+        mesh_area_names = [convert_ras_hdf_string(n) for n in self["/Geometry/2D Flow Areas/Attributes"]["Name"][()]]
+        for mesh_name in mesh_area_names:
+            cell_face_info = self[f"/Geometry/2D Flow Areas/{mesh_name}/Cells Face and Orientation Info"][()]
+            cell_face_values = self[f"/Geometry/2D Flow Areas/{mesh_name}/Cells Face and Orientation Values"][()]
+            cell_cnt = cell_face_info.shape[0]
+            cell_dict["mesh_name"] += [mesh_name]*cell_cnt
+            cell_dict["cell_id"] += range(cell_cnt)
+            cell_dict["face_id_list"] += list(cell_face_values[starting_row:starting_row+count, 0] for starting_row, count in cell_face_info)
+        cell_dict["face_cnt"] = list([len(l) for l in cell_dict["face_id_list"]])
+        cell_df = pd.DataFrame(cell_dict)
+        cell_df = cell_df[cell_df.face_cnt > 1]
+        cell_df["face_coord_list"] = cell_df.apply(lambda x: list([face_dict[str(face_id)+x.mesh_name] for face_id in x.face_id_list]), axis=1)
+        cell_df["geometry"] = cell_df["face_coord_list"].apply(polygonize)
+
+        return GeoDataFrame(cell_df[["mesh_name", "cell_id", "geometry"]], geometry="geometry")
```

Comment:
Should be setting the `crs` here based on the projection extracted from the HDF.
https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.html#geopandas.GeoDataFrame

Reply:
Added crs argument to each GeodataFrame instance -> 6864df8606db67973959a19fb0ec7cb6d947f8ed